src/main/java/com/google/api/generator/gapic/composer/Composer.java

```diff
@@ -100,6 +102,7 @@ private static GapicClass generateGenericClass(Kind kind, String name, Service s
 
     ClassDefinition classDef =
         ClassDefinition.builder()
+            .setFileHeader(ApacheLicenseComposer.APACHE_LICENSE)
```

Comment:
We currently generate many classes, and this method is a placeholder that will be going away. Can you think of ways to restructure this such that we can add a license to a class and test that in isolation? How would we incorporate that into the existing classes?

Reply:
sure, good point. Since we are creating different GapicClasses in the composers, I think a central place to add a file header is `GapicClass` class.  Call `GapicClass.create().addApacheLicnese()` for adding license to existing class. 3 reasons are in consideration:

1. ClassDefinition is another alternative place to have this helper, but in the end, it's gapic class that needs/wraps it, so putting the helper in gapic class might make more sense.
2. The way to incorporate header to the existing class is `GapicClass.create(kind, classDef).addApacheLicense();` 
3. it can be independently tested in `GapicClassTest` without altering composer files.

Alternatives: 
1. tried to have the helper `addApacheLicense(GapicClass)` in `Composer` class, but in that way we will call it by `clazzes.add(**Composer.instance().generate().addApacheLicense())` I think adding file header is more suitable in a class level, instead of composer level.
2. add the helper to `ApacheLicense` as a util, but `ApacheLicense.addLicenseToGapicClass()` method should makes more sense to directly be called from the class side `GapicClass.addApacheLicense()`
Let me know your thoughts @miraleung Thanks

Reply:
GapicClass has no extra logic beyond holding data, so let's keep it that way. Let's add a helper method in Composer.java that adds license headers to each GapicClass instead.

Reply:
updated the `Composer` class and add `ComposerTest` to test file header addition in isolation.